snuba/reader.py

```diff
@@ -70,9 +70,10 @@ def iterate_rows():
     for col in result["meta"]:
         if DATETIME_TYPE_RE.match(col["type"]):
             for row in iterate_rows():
-                row[col["name"]] = (
-                    row[col["name"]].replace(tzinfo=tz.tzutc()).isoformat()
-                )
+                if row[col["name"]] is not None:  # The column value can be null/None at times.
+                    row[col["name"]] = (
+                        row[col["name"]].replace(tzinfo=tz.tzutc()).isoformat()
+                    )
```

Comment:
Ok, good catch, I think we did not experience this before because we had only one nullable datetime column before CDC and that was pretty much never null, also likely we were not grouping by a datetime column with totals (the other way you can get None here).

Before  going on, 
- let's do as Ted said and skip the type conversion only if the column is actually Nullable. It is easy since the regex above is already considering that. The result of the match already contains a `Nullable(` group if the column is nullable
- please do the same for DATE_TYPE_RE (right below)
- please let's cover this case in the unit test: tests/test_api.py::TestApi::test_totals. It should be a trivial addition. if you group by a datetime column you should be stressing this.

thanks